### PR TITLE
Update ContainerRunner to use REST verifier

### PR DESCRIPTION
**Pull request by alexmwu**
We update the LauncherSpec to store the projectID and region.

The ContainerRunner will now be able to use the REST-based verifier.Client, based on the addr flag.
The flag will determine whether the endpoint is the gRPC or REST verifier. The REST verifier is used if the flag is empty or contains "http", otherwise the gRPC verifier is used.

### CODE REVIEW

File: launcher/container_runner.go [hunk 1]
```diff
@@ -207,6 +218,43 @@ func NewRunner(ctx context.Context, cdClient *containerd.Client, token oauth2.To
 	}, nil
 }
 
+// getGRPCClient returns a gRPC verifier.Client pointing to the given address.
+// It also returns a grpc.ClientConn for closing out the connection.
+func getGRPCClient(asAddr string, logger *log.Logger) (verifier.Client, *grpc.ClientConn, error) {
+	opt := grpc.WithTransportCredentials(insecure.NewCredentials())
+	conn, err := grpc.Dial(asAddr, opt)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to open connection to attestation service: %v", err)
```
Comment (josephlr) [inline launcher/container_runner.go:227]: We should mention GRPC in this error message

File: launcher/container_runner.go [hunk 2]
```diff
@@ -207,6 +218,43 @@ func NewRunner(ctx context.Context, cdClient *containerd.Client, token oauth2.To
...
+		return nil, fmt.Errorf("failed to create HTTP client: %v", err)
+	}
+
+	var opts []option.ClientOption
+	opts = append(opts, option.WithHTTPClient(httpClient))
+	if asAddr != "" {
+		opts = append(opts, option.WithEndpoint(asAddr))
+	}
+
+	restClient, err := rest.NewClient(context.Background(),
```
Comment (josephlr) [inline launcher/container_runner.go:247]: feed in `ctx` in stead of `context.Background`

File: launcher/container_runner.go [hunk 3]
```diff
@@ -207,6 +218,43 @@ func NewRunner(ctx context.Context, cdClient *containerd.Client, token oauth2.To
...
+		spec.ProjectID,
+		spec.Region,
```
Comment (josephlr) [inline launcher/container_runner.go:249]: For now, should we just pass `us-central1` as the region? If we do it's fine to keep the `Region` stuff in the launcher spec.

File: launcher/container_runner.go [hunk 4]
```diff
@@ -207,6 +218,43 @@ func NewRunner(ctx context.Context, cdClient *containerd.Client, token oauth2.To
...
+// It defaults to the Attestation Verifier instance at
+// https://confidentialcomputing.googleapis.com.
+func getRESTClient(ctx context.Context, asAddr string, spec spec.LauncherSpec) (verifier.Client, error) {
+	httpClient, err := google.DefaultClient(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create HTTP client: %v", err)
+	}
+
+	var opts []option.ClientOption
+	opts = append(opts, option.WithHTTPClient(httpClient))
```
Comment (josephlr) [inline launcher/container_runner.go:242]: Nit, this can just be:

```go
opts := []option.ClientOption{option.WithHTTPClient(httpClient)}
```

**Review by josephlr — APPROVED** [review]

File: launcher/container_runner.go (hunk 3 above)
Comment (alexmwu) [reply launcher/container_runner.go]: oh sorry you're saying still keep Region in launcher spec because we'll eventually need it. Brainfart on my end.

Comment (josephlr) [reply launcher/container_runner.go]: Right now the REST service is only running in us-central1. Ideally, if we were running this in every region, we would do what your code does, call the attestation verifier endpoint corresponding the Region in which the VM is running.

However, right now, we can only call the service in us-central1. To support the ability to run VM in other regions, we can just have those VMs make a cross-region API call to `us-central1`.

Comment (josephlr) [reply launcher/container_runner.go]: Basically, for now  we just do:
```go
const defaultRegion = "us-central1"
restClient, err := rest.NewClient(ctx, spec.ProjectID, defaultRegion, opts...)
```
but we keep the code that fills out `spec.Region`.

Comment (alexmwu) [reply launcher/container_runner.go]: Done. Thanks!

**Pull request merged**